lasagne/layers/conv.py

```diff
@@ -249,7 +249,7 @@ def get_output_for(self, input, input_shape=None, **kwargs):
                                       filter_shape=self.get_W_shape(),
                                       border_mode='full')
             shift = (self.filter_size[0] - 1) // 2
-            conved = conved[:, :, shift:input.shape[2] + shift]
+            conved = conved[:, :, shift:-shift]
```

Comment:
It won't work for even kernel sizes, but I don't think we were trying to support 'same' for those anyway.

The original way makes more sense to me, but it seems to really confuse Theano.